src/firefly/js/api/ApiUtilChart.jsx

```diff
@@ -0,0 +1,35 @@
+/*
+ * License information at https://github.com/Caltech-IPAC/firefly/blob/master/License.txt
+ */
+import {take} from 'redux-saga/effects';
+
+import {dispatchAddSaga} from '../core/MasterSaga.js';
+import * as TablesCntlr from '../tables/TablesCntlr.js';
+import * as TableUtil from '../tables/TableUtil.js';
+
+import * as XYPlotCntlr from '../visualize/XYPlotCntlr.js';
+//import * as HistogramCntlr from '../visualize/HistogramCntlr.js';
+//import * as TableStatsCntlr from '../visualize/TableStatsCntlr.js';
+
+export {uniqueChartId} from '../visualize/ChartUtil.js';
+
+export function loadPlotDataForTbl(tblId, chartId, xyPlotParams) {
+    dispatchAddSaga(getChart, {tblId, chartId, xyPlotParams});
+}
+
+export function* getChart({tblId, chartId, xyPlotParams}) {
```

Comment:
I think you are right. Just need XYPlotCntlr.LOAD_PLOT_DATA action creator to check if the table is fully loaded. If not, just update the store and  do nothing: syncCharts saga will take care of the server call.